Core/src/org/sleuthkit/autopsy/discovery/ui/DateFilterPanel.java

```diff
@@ -213,26 +247,80 @@ JLabel getAdditionalLabel() {
         return null;
     }
 
+    @Override
+    void addListeners(ActionListener actionListener, ListSelectionListener listListener) {
+        dateFilterCheckBox.addActionListener(actionListener);
+        startCheckBox.addActionListener(actionListener);
+        endCheckBox.addActionListener(actionListener);
+        rangeRadioButton.addActionListener(actionListener);
+        mostRecentRadioButton.addActionListener(actionListener);
+    }
+
+    @Override
+    void removeListeners() {
+        for (ActionListener listener : dateFilterCheckBox.getActionListeners()) {
+            dateFilterCheckBox.removeActionListener(listener);
+        }
+        for (ActionListener listener : rangeRadioButton.getActionListeners()) {
+            rangeRadioButton.removeActionListener(listener);
+        }
+        for (ActionListener listener : mostRecentRadioButton.getActionListeners()) {
+            mostRecentRadioButton.removeActionListener(listener);
+        }
+        for (ActionListener listener : rangeRadioButton.getActionListeners()) {
+            rangeRadioButton.removeActionListener(listener);
+        }
+        for (ActionListener listener : startCheckBox.getActionListeners()) {
+            startCheckBox.removeActionListener(listener);
+        }
+        for (ActionListener listener : endCheckBox.getActionListeners()) {
+            endCheckBox.removeActionListener(listener);
+        }
+    }
+
     @Override
     String checkForError() {
-        return "Domain search is not implemented.";
+        if (dateFilterCheckBox.isSelected()) {
+            if (!(rangeRadioButton.isSelected() || mostRecentRadioButton.isSelected())) {
+                return "Range or Only Last must be selected";
```

Comment:
Should these be bundle messages?

Reply:
Would probably be a good idea, will change.